app/views/layouts/_application_base.html.erb

```diff
@@ -27,6 +27,7 @@
     <div class="wrapper">
       <%= turbo_frame_tag "remote_modal", target: "_top" %>
       <%= yield %>
+      <%= render 'common/accept_cgu_modal' if should_accept_cgu? %>
```

Comment:
Je ne pense pas que ce soit à la vue de gérer cette logique.
Ce serait plus logique d'avoir un `before_action` globale qui vérifie ça.
Aussi je ne pense pas qu'il y ait besoin d'une modale, peut-être qu'une simple page conviendrait mieux. D'ailleurs je crois que la logique qui consiste à mettre le bouton "accepter" en bas des CGU (et donc en forçant à scroller) n'est pas respectée ici.

Reply:
Je suis d'accord sur le before_action 👍 .

En revanche je ne suis pas d'accord sur l'implémentation en tant que page. Une double redirection (dans un sens puis dans l'autre) ne me semble pas du tout être un workflow adapté en terme d'UX car tu perds la vue du contexte, tu ne sais plus où tu reviens, c'est une gymnastique mentale qui me semble violente et inutile. 
Peut-être que Samantha aura un avis là dessus @samanthaauffret 

Je ne suis pas non plus d'accord sur le fait de forcer à scroller. Je ne crois pas qu'il y est un consensus là dessus et comme tu le sais forcer à scroller n'implique pas forcer à lire. Partant de ce constat je ne vois pas l'intérêt de forcer à scroller. Et je ne vois pas non plus en quoi forcer à scroller nous donnerait meilleure garantie que de permettre la lecture comme ici, et faire cocher une case. 

Mon opinion est que ce que l'on veut c'est : 
- les informer qu'on a des CGU
- leur permettre de les lire
- leur demander d'accepter
Le tout en le moins d'étapes possibles. Donc théoriquement on aurait même pu se passer de la case à cocher, mais en terme d'impact mental cela me semble plus fort qu'un simple bouton qu'on peut clicker machinalement. 

Donc je pense que cette implémentation est le meilleur compromis en terme d'UX

Reply:
> Une double redirection (dans un sens puis dans l'autre) ne me semble pas du tout être un workflow adapté en terme d'UX 

Je suis pas sûr de voir ce qui pose problème ici. Un agent se connecte, voit la page des cgu s'il ne l'a pas accepté, puis est redirigé vers la page d'accueil post connexion. Ça me parait raisonnable non ? Surtout que c'est une action qui n'aura lieu qu'une fois.

> Je ne suis pas non plus d'accord sur le fait de forcer à scroller.

Ça vient pas de moi ça c'était ce qui était demandé dans le ticket :).

Reply:
> Pour le coup mon avis est assez tranché sur ce point, certes il ne le voit qu'une fois mais je ne vois vraiment aucune raison d'avoir une page dédié quand on peut afficher ça en modal, ce qui est le cas le plus courant sur internet 

Ok pour la modale alors 👍 

> Après c'est vrai que le ticket est très sujet à interprétation

Oui c'est vrai que c'était pas clair dans le ticket, mais je m'en suis souvenu parce qu'Amaury me l'avait dit en sprint planning (j'aurais peut-être dû le mentionner plus tôt ^^). Je te laisse voir avec lui dans ce cas :).